### PR TITLE
add support for k8s endpoint

### PR DESCRIPTION
**Pull request by k-rister**

### CODE REVIEW

File: .github/actions/integration-tests/base [hunk 1]
```diff
@@ -30,3 +30,29 @@ function log_rc {
...
+
+function validate_ci_run_environment {
+    case "${CI_RUN_ENVIRONMENT}" in
+        standalone|github)
+            echo "CI Run Environment is '${CI_RUN_ENVIRONMENT}'"
+            echo
+            ;;
+        *)
+            echo "ERROR: Unknown value for --run-environment [${CI_RUN_ENVIRONMENT}].  Acceptable values are 'standalone' and 'github'."
+            exit 1
```
Comment (rafaelfolco) [inline .github/actions/integration-tests/base:42]: using same exit code (1)

File: .github/actions/integration-tests/base [hunk 2]
```diff
@@ -30,3 +30,29 @@ function log_rc {
...
+
+function validate_ci_endpoint {
+    case "${CI_ENDPOINT}" in
+        remotehost|k8s)
+            echo "CI Endpoint is '${CI_ENDPOINT}'"
+            echo
+            ;;
+        *)
+            echo "ERROR: Unknown value for --ci-endpoint [${CI_ENDPOINT}].  Acceptable values are 'remotehost' and 'k8s'."
+            exit 1
```
Comment (rafaelfolco) [inline .github/actions/integration-tests/base:55]: not a problem, just not ideal for creating testcases.... if you want to check the desired behavior, you may want to have separate exit codes for each scenario

File: .github/actions/integration-tests/setup-ci-environment [hunk 3]
```diff
@@ -69,6 +70,36 @@ chmod 600 /root/.ssh/authorized_keys
 ssh -o StrictHostKeyChecking=no localhost echo "password-less root login over ssh works"
 stop_github_group
 
+# ensure endpoint availability
+case "${CI_ENDPOINT}" in
+    k8s)
+        start_github_group "Configure k8s endpoint"
```
Comment (rafaelfolco) [inline .github/actions/integration-tests/setup-ci-environment:76]: I still need to get used to this...

File: .github/actions/integration-tests/run-ci-stage1 [hunk 4]
```diff
@@ -242,11 +268,22 @@ run_cmd "podman ps --all --external"
...
 
-run_cmd "podman ps --all --external"
+case "${CI_ENDPOINT}" in
+    k8s)
+        run_cmd "ssh root@localhost kubectl get nodes"
+
+        run_cmd "ssh root@localhost kubectl get pods"
 
-run_cmd "podman images"
+        run_cmd "microk8s ctr images list"
```
Comment (rafaelfolco) [inline .github/actions/integration-tests/run-ci-stage1:277]: this is microk8s specific. If we change to a fedora runner, do we need to change ci again ?

**Review by rafaelfolco — APPROVED** [review]
lgtm

File: .github/actions/integration-tests/run-ci-stage1 (hunk 4 above)
Comment (k-rister) [reply .github/actions/integration-tests/run-ci-stage1:277]: The entire thing is heavily focused on the github runner environment using Ubuntu.  There is a *a lot* that would have to change to support a different runner environment...

**Pull request merged**